### PR TITLE
Use getTransactionReceipt in jsonrpcclient.poll()

### LINKED ISSUE #4497: TypeError: 'NoneType' object is not subscriptable
```
2019-07-31 13:56:19.109921 [error    ] Exception while running scenario [scenario_player.main]
Traceback (most recent call last):
  File "/Users/paul/Projects/brainbot/scenario-player/scenario_player/main.py", line 181, in run
    runner.run_scenario()
  File "/Users/paul/Projects/brainbot/scenario-player/scenario_player/runner.py", line 184, in run_scenario
    gas_limit=mint_gas,
  File "/Users/paul/Projects/brainbot/scenario-player/scenario_player/runner.py", line 248, in _initialize_scenario_token
    token_ctr, token_block = get_or_deploy_token(self)
  File "/Users/paul/Projects/brainbot/scenario-player/scenario_player/utils/legacy.py", line 245, in get_or_deploy_token
    constructor_parameters=(0, decimals, name, symbol),
  File "/Users/paul/Projects/brainbot/scenario-player/venv/lib/python3.7/site-packages/raiden/network/rpc/client.py", line 584, in deploy_single_contract
    contract_address = receipt["contractAddress"]
TypeError: 'NoneType' object is not subscriptable
```

### PR DESCRIPTION
**Pull request by LefterisJP**
Should fix #4497 https://github.com/raiden-network/raiden/issues/3761

### CODE REVIEW

**Comment by LefterisJP** [comment]
@hackaugusto sure

File: raiden/network/proxies/token_network.py [hunk 1]
```diff
@@ -1606,9 +1606,9 @@ def _close(
 
                     # These checks do not have problems with race conditions because
                     # `poll`ing waits for the transaction to be confirmed.
-                    mining_block = int(receipt_or_none["blockNumber"])
+                    mining_block = int(failed_receipt["blockNumber"])
```
Comment (palango) [inline raiden/network/proxies/token_network.py]: I think `blockNumber` is already an int

Comment (LefterisJP) [reply raiden/network/proxies/token_network.py]: perhaps. I just don't want to do any unnecessary changes. But I will try to confirm in a test.

Comment (palango) [reply raiden/network/proxies/token_network.py]: It#s used in other places at least. But we can also do this in a follow up: https://github.com/raiden-network/raiden/pull/4512/files#diff-d02574a7a7eca826412780e80a6326a3R2360

Comment (LefterisJP) [reply raiden/network/proxies/token_network.py]: Yeah checked and it is indeed integer and also only 2 places had the extra int() cast so removed them.

**Pull request merged**